modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/table/Table.tsx

```diff
@@ -290,39 +299,60 @@ const SaveFDSFieldsModalContent = ({
 								</ManagementToolbar.Item>
 
 								<ManagementToolbar.Item className="nav-item-expand">
-									<ClayInput.Group>
-										<ClayInput.GroupItem>
-											<ClayInput
-												insetAfter
-												onChange={(event) =>
-													onSearch(event.target.value)
-												}
-												placeholder={Liferay.Language.get(
-													'search'
-												)}
-												type="text"
-												value={query}
-											/>
-
-											<ClayInput.GroupInsetItem
-												after
-												tag="span"
-											>
-												<ClayButtonWithIcon
-													aria-label={Liferay.Language.get(
-														'search'
-													)}
-													displayType="unstyled"
-													symbol="search"
-												/>
-											</ClayInput.GroupInsetItem>
-										</ClayInput.GroupItem>
-									</ClayInput.Group>
+									<SearchResultsMessage
+										numberOfResults={visibleFields.length}
+									/>
+
+									<AutoSearch
+										onSearch={onSearch}
+										query={query}
+									/>
 								</ManagementToolbar.Item>
 							</ManagementToolbar.ItemList>
 						</ManagementToolbar.Container>
 
-						<div className="bg-light fields pb-2 pt-2">
+						{getSelectedFieldsCount() > 0 && (
+							<ClayResultsBar>
+								<ClayResultsBar.Item expand>
+									<span className="component-text text-truncate-inline">
+										<span className="text-truncate">
+											{getSelectedFieldsCount() > 1
+												? sub(
+														Liferay.Language.get(
+															'x-items-selected'
+														),
+														getSelectedFieldsCount()
+												  )
+												: sub(
+														Liferay.Language.get(
+															'x-item-selected'
+														),
+														getSelectedFieldsCount()
```

Comment:
Rather than calling `getSelectedFieldsCount` multiple times, can you just put it as a variable and only call it once?